Fix build with libavutil 50, ie ffmpeg trunk by not using pixel formats that
have been deprecated for years.


Index: cinelerra-20090210/cinelerra/ffmpeg.C
===================================================================
--- cinelerra-20090210.orig/cinelerra/ffmpeg.C
+++ cinelerra-20090210/cinelerra/ffmpeg.C
@@ -70,11 +70,11 @@ PixelFormat FFMPEG::color_model_to_pix_f
 	switch (color_model) 
 		{ 
 		case BC_YUV422: 
-			  return PIX_FMT_YUV422;
+			  return PIX_FMT_YUYV422;
 		case BC_RGB888:
 			return PIX_FMT_RGB24;
 		case BC_BGR8888:  // NOTE: order flipped
-			return PIX_FMT_RGBA32;
+			return PIX_FMT_RGB32;
 		case BC_BGR888:
 			return PIX_FMT_BGR24;
 		case BC_YUV420P: 
@@ -95,11 +95,11 @@ PixelFormat FFMPEG::color_model_to_pix_f
 int FFMPEG::pix_fmt_to_color_model(PixelFormat pix_fmt) {
 	switch (pix_fmt) 
 		{ 
-		case PIX_FMT_YUV422:
+		case PIX_FMT_YUYV422:
 			return BC_YUV422;
 		case PIX_FMT_RGB24:
 			return BC_RGB888;
-		case PIX_FMT_RGBA32:
+		case PIX_FMT_RGB32:
 			return BC_BGR8888;
 		case PIX_FMT_BGR24:
 			return BC_BGR888;
@@ -286,7 +286,7 @@ int FFMPEG::convert_cmodel(AVPicture *pi
 	// make an intermediate temp frame only if necessary
 	int cmodel_in = pix_fmt_to_color_model(pix_fmt_in);
 	if (cmodel_in == BC_TRANSPARENCY) {
-		if (pix_fmt_in == PIX_FMT_RGBA32) {
+		if (pix_fmt_in == PIX_FMT_RGB32) {
 			// avoid infinite recursion if things are broken
 			printf("FFMPEG::convert_cmodel pix_fmt_in broken!\n");
 			return 1;
Index: cinelerra-20090210/quicktime/qtffmpeg.c
===================================================================
--- cinelerra-20090210.orig/quicktime/qtffmpeg.c
+++ cinelerra-20090210/quicktime/qtffmpeg.c
@@ -225,7 +225,7 @@ static int get_chroma_factor(quicktime_f
 		case PIX_FMT_YUV420P:
 			return 4;
 			break;
-		case PIX_FMT_YUV422:
+		case PIX_FMT_YUYV422:
 			return 2;
 			break;
 		case PIX_FMT_YUV422P:
@@ -424,7 +424,7 @@ int quicktime_ffmpeg_decode(quicktime_ff
 		case PIX_FMT_YUV420P:
 			input_cmodel = BC_YUV420P;
 			break;
-		case PIX_FMT_YUV422:
+		case PIX_FMT_YUYV422:
 			input_cmodel = BC_YUV422;
 			break;
 		case PIX_FMT_YUV422P:
